server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -851,6 +861,161 @@ public Configuration updateConfiguration(final UpdateCfgCmd cmd) throws InvalidP
+        }
+
+        if (scope != null && !scope.equals(ConfigKey.Scope.Global.toString()) && !configScope.contains(scope)) {
+            throw new InvalidParameterValueException("Invalid scope id provided for the parameter " + name);
+        }
+
+        String newValue = null;
+        switch (ConfigKey.Scope.valueOf(scope)) {
+            case Zone:
+                final DataCenterVO zone = _zoneDao.findById(id);
+                if (zone == null) {
+                    throw new InvalidParameterValueException("unable to find zone by id " + id);
+                }
+                _dcDetailsDao.removeDetail(id, name);
+                optionalValue = Optional.ofNullable(configKey.valueIn(id));
+                newValue = optionalValue.isPresent() ? optionalValue.get().toString() : defaultValue;
+                break;
+
+            case Cluster:
+                final ClusterVO cluster = _clusterDao.findById(id);
+                if (cluster == null) {
+                    throw new InvalidParameterValueException("unable to find cluster by id " + id);
+                }
+                ClusterDetailsVO clusterDetailsVO = _clusterDetailsDao.findDetail(id, name);
+                newValue = configKey.value().toString();
+                if (name.equalsIgnoreCase("cpu.overprovisioning.factor") || name.equalsIgnoreCase("mem.overprovisioning.factor")) {
+                    _clusterDetailsDao.persist(id, name, newValue);
+                } else if (clusterDetailsVO != null) {
+                    _clusterDetailsDao.remove(clusterDetailsVO.getId());
+                }
+                optionalValue = Optional.ofNullable(configKey.valueIn(id));
+                newValue = optionalValue.isPresent() ? optionalValue.get().toString() : defaultValue;
+                break;
+
+            case StoragePool:
+                final StoragePoolVO pool = _storagePoolDao.findById(id);
+                if (pool == null) {
+                    throw new InvalidParameterValueException("unable to find storage pool by id " + id);
+                }
+                _storagePoolDetailsDao.removeDetail(id, name);
+                optionalValue = Optional.ofNullable(configKey.valueIn(id));
+                newValue = optionalValue.isPresent() ? optionalValue.get().toString() : defaultValue;
+                break;
+
+            case Domain:
+                final DomainVO domain = _domainDao.findById(id);
+                if (domain == null) {
+                    throw new InvalidParameterValueException("unable to find domain by id " + id);
+                }
+                DomainDetailVO domainDetailVO = _domainDetailsDao.findDetail(id, name);
+                if (domainDetailVO != null) {
+                    _domainDetailsDao.remove(domainDetailVO.getId());
+                }
+                optionalValue = Optional.ofNullable(configKey.valueInDomain(id));
```

Comment:
```suggestion
                optionalValue = Optional.ofNullable(configKey.valueIn(id));
```

Reply:
It should be valueInDomain

Reply:
@ravening jenkins failed

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project cloud-server: Compilation failure: Compilation failure: 
[ERROR] /home/jenkins/jenkins-agent/workspace/Cloudstack/cloudstack-pr-analysis/server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java:[902,41] error: cannot find symbol
[ERROR]   symbol:   class LinkedHashMap
[ERROR]   location: class ConfigurationManagerImpl
[ERROR] /home/jenkins/jenkins-agent/workspace/Cloudstack/cloudstack-pr-analysis/server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java:[974,61] error: cannot find symbol
[ERROR]   symbol:   method valueInDomain(Long)
[ERROR]   location: variable configKey of type ConfigKey<?>
```

Reply:
@weizhouapache yes because, few part of the code are present in another pr